tuning/trackers/aimstack_tracker.py

```diff
@@ -20,6 +24,61 @@
 from .tracker import Tracker
 from tuning.config.tracker_configs import AimConfig
 
+AIM_HASH_EXPORT_FILENAME = "aim_run_hash.json"
+
+
+class CustomAimCallback(AimCallback):
+    """
+    Custom Aimstack callback is used to export run id from Aim
+    as soon as it is created, which is during on_init_end.
+    """
+
+    # path where we export run hash generated by Aim
+    # This is used to link back to the expriments from outside aimstack
+    run_id_export_path = None
+    logger = None
+
+    def on_init_end(self, args, state, control, **kwargs):
+        # pylint: disable=unused-argument
+        """
+        Override Aimstack callback on_init_end function
+        First call AimCallback.setup to initialize internal structures
+        second export Aimstack's run hash to a file
+        hash is exported to, AimConfig.aim_run_id_export_path if it is passed
+        or, training_args.output_dir/aim_run_hash.json if output_dir is present
+        Exported hash looks like '{"run_hash":"<hash>"}' in the file
+        hash is not exported if both paths are invalid
+        """
+        self.setup()  # initialize aim's run_hash
+
+        # Store the run hash
+        # Change default run hash path to output directory
+        if self.run_id_export_path is None:
+            if args and args.output_dir:
+                # args.output_dir/aim_run_hash
+                self.run_id_export_path = os.path.join(
+                    args.output_dir, AIM_HASH_EXPORT_FILENAME
```

Comment:
Do we want a separate file for each tracker? or should we have a tracker.json, which has various sections for each type of tracker?

The info might be different for each, but the process can be same. Also, today it it hash, later we might want to add more.